datafusion/optimizer/src/push_down_limit.rs

```diff
@@ -183,6 +169,65 @@ impl OptimizerRule for PushDownLimit {
     }
 }
 
+/// Wrap the input plan with a limit node
+///
+/// Original:
+/// ```text
+/// input
+/// ```
+///
+/// Return
+/// ```text
+/// Limit: skip=skip, fetch=fetch
+///  input
+/// ```
+fn make_limit(skip: usize, fetch: usize, input: LogicalPlan) -> LogicalPlan {
```

Comment:
I think if you use `input: Arc<LogicalPlan>` here so you can reuse `make_limit` in `make_arc_limit`